generate_directory_indexes.py

```diff
@@ -15,37 +22,68 @@
 
 #sys.setrecursionlimit(2100000000)
 
-parser = argparse.ArgumentParser()
-parser.add_argument("path", help="Path to index and write indexes to")
-parser.add_argument("--noop", help="Only print files to be created without writing them to disk", action="store_true")
-parser.add_argument('--verbose', '-v', help="Verbose output. Repeat (up to -vvv) for more verbosity", action='count')
-args = parser.parse_args()
-
-logger = logging.getLogger(__name__)
-if args.verbose == None:
-    logging.basicConfig(level=logging.ERROR)
-if args.verbose == 1:
-    logging.basicConfig(level=logging.WARNING)
-elif args.verbose == 2:
-    logging.basicConfig(level=logging.INFO)
-else:
-    logging.basicConfig(level=logging.DEBUG)
+script = os.path.basename(sys.argv[0])
```

Comment:
Is there any reason why `script` is in the top-level scope? I'm only seeing it be used in `validate_input`. We could push this down into that function's scope to declutter top-level scope as much as possible.

Reply:
There *was* a reason for it but it's no longer applicable.